charts/rancher-turtles/0.8.0+up0.19.0/templates/metal3-infrastructure.yaml

```
@@ -0,0 +1,69 @@
+{{- if and (index .Values "cluster-api-operator" "cluster-api" "enabled") (index .Values "cluster-api-operator" "cluster-api" "metal3" "enabled") }}
+{{- $namespace := index .Values "cluster-api-operator" "cluster-api" "metal3" "infrastructure" "namespace" }}
+{{- if not (lookup "v1" "Namespace" "" $namespace) }}
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    "helm.sh/hook": "post-install, post-upgrade"
+    "helm.sh/hook-weight": "1"
+  name: {{ index .Values "cluster-api-operator" "cluster-api" "metal3" "infrastructure" "namespace" }}
+{{- end }}
+---
+apiVersion: turtles-capi.cattle.io/v1alpha1
+kind: ClusterctlConfig
+metadata:
+  name: clusterctl-config
+  namespace: rancher-turtles-system
+  annotations:
+    "helm.sh/hook": "post-install, post-upgrade"
+    "helm.sh/hook-weight": "1"
+spec:
```

```suggestion
spec:
  images:
  - name: rancher-fleet
     repository: registry.suse.com/rancher
```
should allow temporary user override for CAAPF image until the next release.

---

This was more of a suggestion. Any patch would work and is only needed until the next turtles release. It was confusing to me to see the `clusterctlconfig` resource as part of the `metal3` infrastructure.

The `clusterctlconfig` resource is a separate resource that can host multiple provider configurations simultaneously, as per [ADR](https://github.com/rancher/turtles/blob/main/docs/adr/0012-clusterctl-provider.md). It essentially acts as a singleton since it’s used by turtles deployed in the `rancher-turtles-system` namespace.